--- modules.d/90dmsquash-live/dmsquash-live-root.orig	2013-11-26 09:14:17.551005743 -0500
+++ modules.d/90dmsquash-live/dmsquash-live-root	2013-11-26 09:16:20.779000017 -0500
@@ -25,17 +25,19 @@
 overlay=$(getarg overlay)

 # FIXME: we need to be able to hide the plymouth splash for the check really
-[ -e $livedev ] & fs=$(blkid -s TYPE -o value $livedev)
+[ -e $livedev ] && fs=$(blkid -s TYPE -o value $livedev)
 if [ "$fs" = "iso9660" -o "$fs" = "udf" ]; then
     check="yes"
 fi
 getarg check || check=""
 if [ -n "$check" ]; then
+   [ -x /bin/plymouth ] && /bin/plymouth --hide-splash
     checkisomd5 --verbose $livedev || :
     if [ $? -ne 0 ]; then
	die "CD check failed!"
	exit 1
     fi
+    [ -x /bin/plymouth ] && /bin/plymouth --show-splash
 fi

 getarg ro && liverw=ro
