--- operatingsystem.rb	2013-07-05 13:05:07.943959513 -0700
+++ operatingsystem.rb	2013-07-05 13:18:54.106022794 -0700
@@ -10,6 +10,8 @@
 #
 # Caveats:
 #
+$LOAD_PATH.unshift('/var/lib/puppet/facts')
+require 'ovirt.rb'

 Facter.add(:operatingsystem) do
   confine :kernel => :sunos
@@ -25,8 +27,13 @@
 Facter.add(:operatingsystem) do
   confine :kernel => :linux
   setcode do
-    if Facter.value(:lsbdistid) == "Ubuntu"
-       "Ubuntu"
+    if FileTest.exists?("/etc/default/version")
+      txt = File.read("/etc/default/version")
+      if txt =~ /^PRODUCT='(.*?)'/
+          $1.gsub(/\s/, '')
+      end
+    elsif Facter.value(:lsbdistid) == "Ubuntu"
+      "Ubuntu"
     elsif FileTest.exists?("/etc/debian_version")
       "Debian"
     elsif FileTest.exists?("/etc/gentoo-release")
